src/v/test_utils/profile_utils.cc

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "test_utils/profile_utils.h"
+
+#include "container/lw_shared_container.h"
+#include "redpanda/admin/api-doc/debug.json.hh"
+#include "resource_mgmt/cpu_profiler.h"
+#include "resource_mgmt/memory_sampling.h"
+
+#include <seastar/core/file.hh>
+#include <seastar/core/fstream.hh>
+#include <seastar/core/sharded.hh>
+#include <seastar/http/httpd.hh>
+#include <seastar/json/json_elements.hh>
+#include <seastar/util/log.hh>
+
+#include <functional>
+#include <limits>
+#include <mutex>
+#include <optional>
+
+namespace test_utils {
+
+class profile_helper {
+public:
+    explicit profile_helper(
+      std::chrono::milliseconds cpu_profiler_sample_period)
+      : _sample_period(cpu_profiler_sample_period) {}
+
+    ss::future<> start();
+    ss::future<> stop();
+
+    ss::future<> write_cpu_profiler_results(std::filesystem::path path);
+    ss::future<> write_memory_profiler_results(std::filesystem::path path);
+
+private:
+    std::chrono::milliseconds _sample_period;
+    ss::sharded<resources::cpu_profiler> _cpu_profiler;
+    ss::sharded<memory_sampling> _memory_sampler;
+    seastar::logger _logger{"profile_helper_logger"};
+};
+
+ss::future<> profile_helper::start() {
+    co_await _cpu_profiler.start(
```

Comment:
any reason you didn't use the plain seastar APIs here?

Reply:
For the memory sampler I guess there isn't much reason not to use the seastar APIs. However, for the cpu profiler there is some logic in the service that I didn't want to replicate in this. I.e, setting up a timer to poll for samples from seastar. I'll stop using the memory sampler service here and use the APIs directly.